From 955661ef88443c43858c1ad246be38d205c5d5fe Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Tue, 8 Aug 2017 14:53:08 +0800
Subject: [PATCH 181/184] octeon-hcd.c: remove reference on IRQF_DISABLED

The IRQF_DISABLED is a NOOP and scheduled to be removed.  According to Ingo
Molnar in commit e58aa3d2d0cc01ad8d6f7f640a0670433f794922 (genirq: Run irq
handlers with interrupts disabled), running IRQ handlers with interrupts
enabled can cause stack overflows when the interrupt line of the issuing
device is still active.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/usb/host/octeon-hcd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/octeon-hcd.c b/drivers/usb/host/octeon-hcd.c
index 2b8f815..3cbae6c 100644
--- a/drivers/usb/host/octeon-hcd.c
+++ b/drivers/usb/host/octeon-hcd.c
@@ -809,7 +809,7 @@ static int octeon_usb_probe(struct platform_device *pdev)
 	cvmx_usb_poll(&priv->usb);
 	spin_unlock_irqrestore(&priv->lock, flags);
 
-	status = usb_add_hcd(hcd, irq, IRQF_DISABLED);
+	status = usb_add_hcd(hcd, irq, 0);
 	if (status) {
 		DEBUG_FATAL("OcteonUSB: USB add HCD failed with %d\n", status);
 		kfree(hcd);
-- 
1.9.1

